lms/templates/dashboard/_dashboard_course_listing.html

```diff
@@ -30,20 +31,24 @@
 
 <li class="course-item">
   % if settings.FEATURES.get('ENABLE_VERIFIED_CERTIFICATES'):
-    % if enrollment.mode == "verified":
-      % if verification_status.get('status') in [VERIFY_STATUS_NEED_TO_VERIFY, VERIFY_STATUS_SUBMITTED, VERIFY_STATUS_APPROVED]:
-        <% mode_class = " verified" %>
-      % else:
-        <% mode_class = " honor" %>
-      % endif
-    % else:
-      <% mode_class = " " + enrollment.mode %>
-    % endif
+    <% course_verified_certs = CourseMode.get_certificate_display(enrollment.mode, verification_status.get('status')) %>
+    <%
+        display_mode = course_verified_certs.get('display_mode')
+        if display_mode == 'verified':
+             mode_class = 'verified'
+        elif display_mode == 'honor':
+            mode_class = 'honor'
+        elif display_mode == 'professional':
+            mode_class = 'professional'
+        else:
```

Comment:
Looks like one test case is still failing because this isn't handling 'audit' correctly.  (You can [read this wiki page](https://openedx.atlassian.net/wiki/display/ECOM/How+To+Set+Up+Course+Modes) for an explanation of what audit mode is.

It looks like this if/else block is basically setting `mode_class` to `display_mode`.  Why not just use `display_mode` as the CSS class?